flow/sequencer.go

```diff
@@ -0,0 +1,135 @@
+package flow
+
+import (
+	"context"
+	"time"
+
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/macformula/hil/utils"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+const (
+	_loggerName = "sequencer"
+)
+
+// Sequencer is responsible for managing the setup and execution of a Sequence.
+type Sequencer struct {
+	l            *zap.Logger
+	progress     Progress
+	progressFeed event.Feed
+
+	fatalErr *utils.ResettableError
+}
+
+// NewSequencer returns a Sequencer object reference.
+func NewSequencer(l *zap.Logger) *Sequencer {
+	return &Sequencer{
+		l:            l.Named(_loggerName),
+		progressFeed: event.Feed{},
+		fatalErr:     utils.NewResettaleError(),
+	}
+}
+
+// SubscribeToProgress subscribes to the progress of the Sequencer accross its Sequence runs.
+// The Progress channel gets updated whenever there is new information available.
+func (s *Sequencer) SubscribeToProgress(progCh chan Progress) event.Subscription {
```

Comment:
Does there need to be an unsubscribe as well? I've never used ethereum events so I'm not sure

Reply:
The subscription it returns allows you to unsubscribe